python/pyspark/_typing.pyi

```diff
@@ -17,17 +17,27 @@
 # under the License.
 
 from typing import Callable, Iterable, Sized, TypeVar, Union
-from typing_extensions import Protocol
+from typing_extensions import Literal, Protocol
+
+from numpy import int32, int64, float32, float64, ndarray
```

Comment:
I wonder how it works when users actually import these codes. `numpy` is technically an optional dependency in RDD API at PySpark. Would this cause to fail in the middle due to missing numpy import by any change?

Reply:
It is safe. This is still a stub file, so at the moment it is not even importable on runtime (and all the imports are handled in `if TYPE_CHECKING` blocks).

If we ever want to migrate this to type `.py`, it might be a problem, but so will be `typing_extensions`.